docs/configuration/rbac.md

```diff
@@ -4,252 +4,449 @@ title: RBAC
+            prefix: "" # optional prefix for all group names; useful especially for channel name mapping
+```
+
+### Mapping types
 
-Each executor and source plugin can provide a `context` section with `rbac` config.
-This config is used to generate a dedicated kube config.
+For both user and group, the following mapping types are supported:
 
-Supported mapping:
+- `Static`
 
-- **Static** mapping - user or group impersonation, always the same subject for given plugin
-- **Channel** mapping - name of the channel is used as subject for group impersonation, only available for executor plugins
+  For user, it uses a single static value. For group, it uses a list of static values. The value is prepended with an optional prefix.
 
-:::note
-Automated actions only support **Static** mapping.
-:::
+- `ChannelName`
 
-#### Example - kubectl executor with read-only RBAC
+  Channel name is used as subject for user or group impersonation. The channel name is prepended with an optional prefix.
+  This mapping is only available for executor plugins.
 
-In this example a single executor plugin is defined with static RBAC that maps to user `kubectl-read-only`.
+### Default configuration
 
-1. ClusterRole and ClusterRoleBinding
+When a given plugin have `context.rbac` property undefined, Botkube doesn't generate a kubeconfig for this plugin. To request kubeconfig generation, define `context.rbac` property with empty value:
 
 ```yaml
-  apiVersion: rbac.authorization.k8s.io/v1
-  kind: ClusterRole
-  metadata:
-    name: kubectl-read-only
-  rules:
-  - apiGroups: ["*"]
-    resources: ["*"]
-    verbs: ["get", "watch", "list"]
-  ---
-  apiVersion: rbac.authorization.k8s.io/v1
-  kind: ClusterRoleBinding
-  metadata:
-    name: kubectl-read-only
-  roleRef:
-    apiGroup: rbac.authorization.k8s.io
-    kind: ClusterRole
-    name: kubectl-read-only
-  subjects:
-  - kind: User
-    name: kubectl-read-only
-    apiGroup: rbac.authorization.k8s.io
+executors:
+  "kubectl":
+    botkube/kubectl@v1:
+      enabled: true
+      # ...
+      context:
+        rbac: {} # enable kubeconfig generation
 ```
 
-Here we define a plugin with _Static_ mapping to **User.rbac.authorization.k8s.io** `kubectl-read-only`.
+However, such configuration will generate a kubeconfig with empty impersonation config, which effectively means an anonymous access to the Kubernetes API.
 
-2. Executor definition with RBAC
+During Botkube installation, Botkube generates Kubernetes ClusterRole and ClusterRoleBinding resources with read-only access for the default group `botkube-plugins-default`. This group is used by default across the `values.yaml` for all default plugins.
+
+```yaml
+rbac:
+  # ...
+  create: true
+  rules: # rules for the default static group
+    - apiGroups: ["*"]
+      resources: ["*"]
+      verbs: ["get", "watch", "list"]
+  staticGroupName: &static-group-name "botkube-plugins-default" # group name used in the `values.yaml` file for default plugins
```

Comment:
```suggestion
  staticGroupName: "botkube-plugins-default" # group name used in the `values.yaml` file for default plugins
```

Reply:
I thought about it, and initially removed the anchor, but then I restored it to be closer to the `values.yaml` content 🤔 However I don't have strong opinion here, so I'll remove it 👍